lab-adam/test/city-route-test.js

```diff
@@ -0,0 +1,295 @@
+          if (err) return done(err);
+          expect(res.status).to.equal(200);
+          for (var i in updateCity) {
+            if (i !== '_id') expect(res.body[i]).to.equal(updateCity[i]);
+          }
+          this.tempCity = res.body;
+          done();
+        });
+      });
+    });
+    describe('with invalid or no body', function() {
+      before(done => {
+        exampleCity.timestamp = new Date();
+        new City(exampleCity).save()
+        .then(city => {
+          this.tempCity = city;
+          done();
+        })
+        .catch(done);
+      });
+      after(done => {
+        delete exampleCity.timestamp;
+        if(this.tempCity){
+          City.remove({})
+          .then(() => done())
+          .catch(done);
+          return;
+        }
+        done();
+      });
+      it('should respond with bad request & 400 error', done => {
+        request.put(`${url}/api/city/${this.tempCity.id}`)
+        .set('Content-Type','application/json')
+        .send('{')
+        .end((err, res) => {
+          expect(res.status).to.equal(400);
+          done();
+        });
+      });
+    });
+    describe('with invalid ID', function() {
+      it('should return a 404', done => {
+        request.put(`${url}/api/city/1234512312`)
+        .end((err, res) => {
+          expect(res.status).to.equal(404);
+          done();
+        });
+      });
+    });
+  });
+  describe('testing POST requests', function(){
+    describe('with valid body', function(){
+      after(done => {
+        if(this.tempCity){
+          City.remove({})
+          .then(() => done())
+          .catch(done);
+          return;
+        }
+        done();
+      });
+      it('should return a city', done => {
+        request.post(`${url}/api/city`)
+        .send(exampleCity)
+        .end((err, res) => {
+          expect(res.status).to.equal(200);
+          for (var i in exampleCity) {
+            if (i === 'id') continue;
+            expect(res.body.i).to.equal(exampleCity.i);
+          }
+          this.tempCity = res.body;
+          done();
+        });
+      });
+    });
+    describe('with invalid or no body', function(){
+      after(done => {
+        if(this.tempCity){
+          City.remove({})
+          .then(() => done())
+          .catch(done);
+          return;
+        }
+        done();
+      });
+      it('should return a 400 error', done => {
+        request.post(`${url}/api/city`)
+        .send({population:'not a Number'})
+        .end((err, res) => {
+          expect(res.status).to.equal(400);
+          done();
+        });
+      });
+    });
+  });
+  describe('testing DELETE requests', function(){
+    describe('with valid body', function(){
+      before(done => {
+        exampleCity.timestamp = new Date();
+        new City(exampleCity).save()
+        .then(city => {
+          this.tempCity = city;
+          done();
+        })
+        .catch(done);
+      });
+      after(done => {
```

Comment:
You shouldn't need to delete anything from your DB in an after block if your delete request works.